sdk/textanalytics/ai-text-analytics/src/textAnalyticsClient.ts

```diff
@@ -354,7 +354,16 @@ export class TextAnalyticsClient {
         code: CanonicalCode.UNKNOWN,
         message: e.message
       });
-      throw e;
+      if (e.response.parsedBody.error.innererror.code === "InvalidDocumentBatch") {
+        const innerError = {
+          statusCode: e.statusCode,
+          code: e.response.parsedBody.error.innererror.code,
+          message: e.response.parsedBody.error.innererror.message
+        };
+        throw innerError;
+      } else {
+        throw e;
+      }
```

Comment:
@willmtemple does that look reasonable to you?

Reply:
You probably also need to set the message in the span. so instead of `message: e.message` L355, you'd need to get the innerError's message beforehand.

Instead of throwing in two codepaths, I would suggest finding the true values of code, message, and statusCode first, then letting the previously-existing error-handling code handle it the same way it always used to.

Reply:
done.